### PR TITLE
Rewrite supported browsers summary, update Safari support info

### PR DESCRIPTION
**Pull request by fvsch**
# PR Description

This PR fixes the issue number: n.a.

### Summary of my changes and explanation of my decisions

- Update the Safari support info to say “Safari 16.4” instead of “Safari 16.4 TP”, since Safari 16.4 shipped.
- Rewrite the summary and add a browser support matrix listing supported desktop and mobile browsers.
- Tweaks to support info for Firefox and Safari.

With the added browser support matrix at the top, there is a bit of duplication with the per-browser entries lower down in the page. I don't know if we consider this a problem or not. Readers who jump to a specific browser and those who list the list at the top would both get correct information, so maybe the duplication works?

Finally, I’ve added some general information about mobile browser support, because we had none. The only suggestion of our mobile browser support was implicit in this sentence:

> we support desktop Chromium-based browsers out of the box

We initially didn't support mobile browsers at all, but I think we currently have decent support on Android, and experimental support in Safari on iOS. I tried to convey this, but definitely would love feedback from the WebContainers team.

### CODE REVIEW

File: docs/platform/webcontainers/browser-support.md [hunk 1]
```diff
@@ -56,20 +71,19 @@ Learn [how to configure Brave to run WebContainers](/platform/webcontainers/brow
...
+Safari does not fully support the required mode for cross-origin isolation, so you might encounter limitations when **running a server in a preview frame** within the StackBlitz editor. Third-party assets might get blocked due to the limitations of cross-origin isolation policies. However, you can work around this by opening your server preview in a separate window.
 
-* [Atomics.waitAsync](https://github.com/tc39/proposal-atomics-wait-async)
-* [Lookbehind in regular expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Assertions)
+Because Safari and Node.js use different JavaScript engines (JavaScriptCore and V8), there is a small risk of running into [JS engine differences](#js-engine-differences).
 
-These cannot be pollyfilled.
+Older Safari versions are not suported. They lack a few necessary Web Platform features, such as [`Atomics.waitAsync`](https://github.com/tc39/proposal-atomics-wait-async) and [lookbehind in regular expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Assertions), which cannot be polyfilled.
```
Comment (fvsch) [inline docs/platform/webcontainers/browser-support.md:86]: Rewrote the Safari section.

- Structure and language is similar to the Firefox section.
- Minimized the information about features missing in Safari (now that it's about older versions of Safari, we can de-emphasize it).

File: docs/platform/webcontainers/browser-support.md [hunk 2]
```diff
@@ -12,25 +12,40 @@ head:
 
 _Last update: February 2023_
 
-**TL;DR** For WebContainers, we support desktop Chromium-based browsers out of the box, as well as Safari 16.4 TP and Firefox, which both are in beta. If you have issues with supported browsers, [check your browser configuration](/platform/webcontainers/browser-config).
+## Supported browsers
 
-:::warning Note
-There is a reported Chrome regression on Macbooks with M1 chip, which also affects the speed of some larger projects on WebContainers. Learn more about this issue in these bug reports: [issue 1228686](https://bugs.chromium.org/p/chromium/issues/detail?id=1228686) and [issue 1356099](https://bugs.chromium.org/p/chromium/issues/detail?id=1356099).
-:::
```
Comment (fvsch) [inline docs/platform/webcontainers/browser-support.md:19]: I have a separate PR for removing the M1 warnings (https://github.com/stackblitz/docs/pull/106), but wanted to see the result of these page updates without those warnings too.

File: docs/platform/webcontainers/browser-support.md [hunk 3]
```diff
@@ -12,25 +12,40 @@ head:
...
 
-However, for cross-origin isolation to work for our use case, you need to be able to embed arbitrary resources: to be able to write and test your web application seamlessly, regardless of which images or scripts you choose to include. For this to work, a [new mode](https://github.com/WICG/credentiallessness) of cross-origin isolation that allows this is needed.
+However, for cross-origin isolation to work for our use case, you need to be able to embed arbitrary resources: to be able to write and test your web application seamlessly, regardless of which images or scripts you choose to include. For this to work, a [new `credentialless` mode](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Embedder-Policy#credentialless) of cross-origin isolation that allows this is needed.
```
Comment (fvsch) [inline docs/platform/webcontainers/browser-support.md:42]: Changed the link to `credentialless` from the WICG repo (https://github.com/WICG/credentiallessness) to a MDN page, because the proposal was merged in the HTML and Fetch specifications.

MDN doesn't have a lot of places or details we can link to, this was the best fit I could find at this time. (There are pages about `<iframe credentialless>` and `window.credentialless` but those are different features.)

File: docs/platform/webcontainers/browser-support.md [hunk 4]
```diff
@@ -12,25 +12,40 @@ head:
 
 _Last update: February 2023_
 
-**TL;DR** For WebContainers, we support desktop Chromium-based browsers out of the box, as well as Safari 16.4 TP and Firefox, which both are in beta. If you have issues with supported browsers, [check your browser configuration](/platform/webcontainers/browser-config).
+## Supported browsers
```
[the remark this answers is not in the document]
Comment (sylwiavargas) [reply docs/platform/webcontainers/browser-support.md:15]: It definitely is a good title.

File: docs/platform/webcontainers/browser-support.md [hunk 5]
```diff
@@ -12,25 +12,40 @@ head:
...
+
+- Chrome: full support.
+- Other Chromium-based browsers: full support. Some browsers may apply restrictive rules by default, and require specific configuration, like [Brave](#brave).
+- Firefox: beta support ([see details](#firefox)).
+- Safari: beta support since Safari 16.4, no support for earlier versions ([see details](#safari)).
+
+### Mobile browsers
+
+- Chrome, Chromium-based browsers and Firefox on Android: beta support. Depending on your device, large projects may run into memory limitations.
+- Safari on iOS 16.4: experimental support. Many projects may run into memory limitations and other compatibility issues.
```
Comment (d3lm) [inline docs/platform/webcontainers/browser-support.md:29]: Not yet supported though. I mean the editor will launch but WebContainer won't boot until we have a separate build for iOS and iPadOS. Maybe it also makes sense to separate iOS and iPadOS as there is overlap but the memory restrictions are different between both and tablets have a little higher memory limits. iOS will be a lot more restrictive and I don't think iOS is something we should support because almost no project with dependencies will run, and it would also mean we'd need a build for iOS and another one for iPadOS. Plus I think we can't even distinguish between the two via the user agent, can we?

**Review by d3lm — COMMENTED** [review]
Nice 👏

File: docs/platform/webcontainers/browser-support.md [hunk 6]
```diff
@@ -12,25 +12,39 @@ head:
...
+
+- Chrome: full support.
+- Other Chromium-based browsers: full support. Some browsers may apply restrictive rules by default, and require specific configuration, like [Brave](#brave).
+- Firefox: beta support ([see details](#firefox)).
+- Safari: beta support since Safari 16.4, no support for earlier versions ([see details](#safari)).
+
+### Mobile browsers
+
+- Android: beta support for Chrome, Chromium-based browsers and Firefox. Depending on your device, large projects may run into memory limitations.
+- iOS and iPadOS: not supported yet. WebContainers can theoretically work on iOS 16.4 and later, but memory limitations for web apps in Safari for iOS and iPadOS are currently too strict to run most WebContainers-based projects.
```
Comment (fvsch) [inline docs/platform/webcontainers/browser-support.md:29]: @d3lm I rewrote the info about iOS. Does that look good, or do we want to rework it a bit?

File: docs/platform/webcontainers/browser-support.md (hunk 5 above)
Comment (samdenty) [reply docs/platform/webcontainers/browser-support.md]: we can distinguish via user agents as ipad gives a desktop UA. IOS gives a mobile UA. We can check for touch support to differentiate between desktop safari and ipad safari

Comment (d3lm) [reply docs/platform/webcontainers/browser-support.md]: Yea we may need to be a little more clever if it's possible to detect. I haven't looked into it but if you say we can then that'd be good. But we do want to mention the differences in memory restrictions between iPad and iOS.

File: docs/platform/webcontainers/browser-support.md (hunk 6 above)
Comment (d3lm) [reply docs/platform/webcontainers/browser-support.md:29]: Yep I think that is accurate as of right now, at least until we test and possibly deploy a separate build for iPadOS.

File: docs/platform/webcontainers/browser-support.md [hunk 7]
```diff
@@ -12,25 +12,39 @@ head:
...
 
-`SharedArrayBuffer`s (SABs) allow simultaneous access to a chunk of memory from multiple different workers. This is a powerful feature that was [disabled temporarily](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer#security_requirements) in light of potential security issues. Cross-origin isolation is the key to enabling SABs: by properly configuring some of the headers and controlling which resources are served to browsers, a site can be considered `crossOriginIsolated` or, in other words, secure enough to use SABs. Both features are enabled in Chromium-based browsers (Chrome, Brave, Edge), Safari 16.4 TP, and Firefox.
+Cross-origin isolation is the key to enabling SABs: by properly configuring some of the headers and controlling which resources are served to browsers, a site can be considered [`crossOriginIsolated`](https://developer.mozilla.org/en-US/docs/Web/API/crossOriginIsolated) or, in other words, secure enough to use SABs. Both features are enabled in Chromium-based browsers (Chrome, Brave, Edge, etc.), Safari 16.4, and Firefox.
```
Comment (sylwiavargas) [inline docs/platform/webcontainers/browser-support.md:39]:
```suggestion
Cross-origin isolation is the key to enabling SABs: by properly configuring some of the headers and controlling which resources are served to browsers, a site can be considered [`crossOriginIsolated`](https://developer.mozilla.org/en-US/docs/Web/API/crossOriginIsolated) or, in other words, secure enough to use SABs. Both features are enabled in Chromium-based browsers (Chrome, Brave, Edge, among others), Safari 16.4, and Firefox.
```

removing latinisms

File: docs/platform/webcontainers/browser-support.md [hunk 8]
```diff
@@ -56,20 +70,19 @@ Learn [how to configure Brave to run WebContainers](/platform/webcontainers/brow
 
 We have beta support for Firefox. Please try it and [share feedback with us](https://github.com/stackblitz/webcontainer-core/issues/new/choose)!
 
-As mentioned above, Firefox does not fully support the required mode for cross-origin isolation, so you might encounter limitations when **running a server in a preview window** within the StackBlitz editor. Third-party assets might get blocked due to the limitations of cross-origin isolation policies. However, you can work around this by opening your server preview in a separate window.
+Firefox does not fully support the required mode for cross-origin isolation, so you might encounter limitations when **running a server in a preview frame** within the StackBlitz editor. Third-party assets might get blocked due to the limitations of cross-origin isolation policies. However, you can work around this by opening your server preview in a separate window.
 
-In addition to this, there might be [other runtime incompatibilities](#runtime-differences) as detailed above.
+Because Firefox and Node.js use different JavaScript engines (SpiderMonkey and V8), there is a small risk of running into [JS engine differences](#js-engine-differences).
```
Comment (sylwiavargas) [inline docs/platform/webcontainers/browser-support.md:75]:
```suggestion
Because Firefox and Node.js use different JavaScript engines (SpiderMonkey and V8, respectively), there is a small risk of running into [JS engine differences](#js-engine-differences).
```

File: docs/platform/webcontainers/browser-support.md [hunk 9]
```diff
@@ -56,20 +70,19 @@ Learn [how to configure Brave to run WebContainers](/platform/webcontainers/brow
...
 
-Starting with Safari 16.4 TP, WebContainers are in beta for Safari.
+We have beta support for Safari on macOS, starting with Safari 16.4. Please try it and [report issues here](https://github.com/stackblitz/webcontainer-core/issues/new/choose).
 
-Older versions of the browser are still lacking a few other features necessary for WebContainers to run, such as:
+Safari does not fully support the required mode for cross-origin isolation, so you might encounter limitations when **running a server in a preview frame** within the StackBlitz editor. Third-party assets might get blocked due to the limitations of cross-origin isolation policies. However, you can work around this by opening your server preview in a separate window.
 
-* [Atomics.waitAsync](https://github.com/tc39/proposal-atomics-wait-async)
-* [Lookbehind in regular expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Assertions)
+Because Safari and Node.js use different JavaScript engines (JavaScriptCore and V8), there is a small risk of running into [JS engine differences](#js-engine-differences).
```
Comment (sylwiavargas) [inline docs/platform/webcontainers/browser-support.md:83]:
```suggestion
Because Safari and Node.js use different JavaScript engines (JavaScriptCore and V8, respectively), there is a small risk of running into [JS engine differences](#js-engine-differences).
```

File: docs/platform/webcontainers/browser-support.md (hunk 1 above)
Comment (sylwiavargas) [reply docs/platform/webcontainers/browser-support.md:85]: awesome

**Review by sylwiavargas — APPROVED** [review]
Thank you, Florens - this looks good.
I've added a few comments and I trust your judgement on them.

File: docs/platform/webcontainers/browser-support.md [hunk 10]
```diff
@@ -12,25 +12,39 @@ head:
...
 
 ## Web Platform requirements
 
-StackBlitz requires some of the latest additions to the Web Platform to work correctly when running WebContainers-based projects. Most important among them are **[`SharedArrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer)** and **[cross-origin isolation](https://developer.mozilla.org/en-US/docs/Web/API/crossOriginIsolated)**.
+StackBlitz requires some of the latest additions to the Web Platform to work correctly when running WebContainers-based projects. Most important among them are **`SharedArrayBuffer`** and **cross-origin isolation**.
+
+### Cross-origin isolation
+
+[`SharedArrayBuffer`s](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer) (SABs) allow simultaneous access to a chunk of memory from multiple different workers. This is a powerful feature that was [disabled temporarily](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer#security_requirements) in light of potential security issues.
```
[the remark this answers is not in the document]
Comment (fvsch) [reply docs/platform/webcontainers/browser-support.md:37]: Not sure this works, because the way it was addressed was by adding the COOP/COEP requirements. So they're still disabled by defaults and will always be.
